### PR TITLE
feat: structs for server's map

### PR DESCRIPTION
**Pull request by theophile-jr**
Feature the structures and functions for the servers map.

### CODE REVIEW

**Review by Njord201 — APPROVED** [review]

File: server/include/map/map.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    int phiras;
+    int thystame;
+} tile_t;
+
+/**
+* @brief Get the map object.
+*
+* @param width
+* @param height
+* @return tile_t***
```
Comment (AxelF44) [inline server/include/map/map.h:45]: Document your parameters

File: server/include/map/map.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+*/
+tile_t ***get_map(int width, int height);
+
+/**
+* @brief Get the tile object.
+*
+* @param map
+* @param height
+* @param width
```
Comment (AxelF44) [inline server/include/map/map.h:54]: Document your parameters

File: server/include/map/map.h [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+*/
+void free_map(tile_t ***map, int height, int width);
+
+/**
+* @brief Display the map object.
+*
+* @param map
+* @param height
+* @param width
```
Comment (AxelF44) [inline server/include/map/map.h:63]: Document your parameters

File: server/src/main.c [hunk 4]
```diff
@@ -11,20 +11,22 @@
 #include "app/app.h"
 #include "ai/team.h"
 #include "parsing.h"
-#include "list/list.h"
+#include "map/map.h"
```
Comment (AxelF44) [inline server/src/main.c:14]: Put your include in ascending order

**Review by AxelF44 — CHANGES_REQUESTED** [review]

File: server/src/map/map.c [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+** map
+*/
+
+#include "map/map.h"
+#include "stdlib.h"
+#include "stdio.h"
+
+tile_t ***get_map(int width, int height)
+{
+    tile_t ***map = malloc(sizeof(tile_t **) * height);
```
Comment (mathieurobert1) [inline server/src/map/map.c:14]: check your malloc

File: server/src/map/map.c [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+    for (int i = 0; i < height; i++) {
+        map[i] = malloc(sizeof(tile_t *) * width);
```
Comment (mathieurobert1) [inline server/src/map/map.c:17]: Check your malloc

File: server/src/map/map.c [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        for (int j = 0; j < width; j++) {
+            map[i][j] = malloc(sizeof(tile_t));
```
Comment (mathieurobert1) [inline server/src/map/map.c:19]: check your malloc

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]
Be careful, check your mallocs.

**Review by AxelF44 — APPROVED** [review]

**Review by Njord201 — APPROVED** [review]

**Review by mathieurobert1 — APPROVED** [review]
Nice job

**Pull request merged**